### PR TITLE
Add `ns_exponent` parameter to control the negative sampling distribution for `*2vec` models. Fix #2090

### LINKED ISSUE #2090: Gensim doesn't allow changing negative sampling distribution parameter
#### Description
Like pointed out in the following article, the negative sampling distribution parameter, which is fixed as 0.75 in Gensim, is worth tuning, specially for other applications beyond NLP. So, I'd be very helpful to make it a parameter for the Word2Vec, instead of fixing it.

https://arxiv.org/abs/1804.04212

### PR DESCRIPTION
**Pull request by fernandocamargoti**
Fixes #2090

Description:

Like pointed out in the following article, the negative sampling distribution parameter, which is fixed as 0.75 in Gensim, is worth tuning, specially for other applications beyond NLP. So, I'd be very helpful to make it a parameter for the Word2Vec, instead of fixing it.

https://arxiv.org/abs/1804.04212

### CODE REVIEW

File: gensim/models/doc2vec.py [hunk 1]
```diff
@@ -325,6 +325,11 @@ def __init__(self, documents=None, dm_mean=None, dm=1, dbow_words=0, dm_concat=0
             If > 0, negative sampling will be used, the int for negative specifies how many "noise words"
             should be drawn (usually between 5-20).
             If set to 0, no negative sampling is used.
+        ns_exponent : float
+            The exponent used to smooth the cumulative distribution used for negative sampling.
+            1.0 leads to a sampling based on the frequency distribution, 0.0 makes items beings sampled equally,
+            while a negative value makes unpopular items being sampled more often than popular onces. The default value
```
Comment (gojomo) [inline gensim/models/doc2vec.py]: For clarity, grammar, and to give a hint of when this could be beneficially tuned, I'd reword as:

"The exponent used to shape the negative sampling distribution. A value of 1.0 samples exactly in proportion to the frequencies, 0.0 samples all words equally, while a negative value samples low-frequency words more than high-frequency words. The popular default value of 0.75 was chosen by the original Word2Vec paper. More recently, in <https://arxiv.org/abs/1804.04212>, Caselles-Dupré, Lesaint, & Royo-Letelier suggest that other values may perform better for recommendation applications."

File: gensim/models/fasttext.py [hunk 2]
```diff
@@ -210,6 +210,11 @@ def __init__(self, sentences=None, sg=0, hs=0, size=100, alpha=0.025, window=5,
             If > 0, negative sampling will be used, the int for negative specifies how many "noise words"
             should be drawn (usually between 5-20).
             If set to 0, no negative sampling is used.
+        ns_exponent : float
+            The exponent used to smooth the cumulative distribution used for negative sampling.
+            1.0 leads to a sampling based on the frequency distribution, 0.0 makes items beings sampled equally,
+            while a negative value makes unpopular items being sampled more often than popular onces. The default value
+            is empirically set to 0.75 following the original paper of Word2Vec.
```
Comment (gojomo) [inline gensim/models/fasttext.py]: Same as above.

**Comment by gojomo** [comment]
Code looks good; I've suggested rewording the comment for clarity & to give a hint/pointer why this might be changed.

**Comment by fernandocamargoti** [comment]
Hello, @gojomo. I've made the adjustments. Thank you for preparing a better text to document this parameter.

**Comment by fernandocamargoti** [comment]
It seems the build failed with some kind of timeout when running the tests for Python 3.5. After all, it was passing before I simply changed the docs.

**Comment by gojomo** [comment]
Looks like it was a spurious failure unrelated to your commits; I forced a retry and it succeeded.

**Comment by menshikh-iv** [comment]
LGTM, @fernandocamargoti please resolve merge conflict and I'll merge your PR

**Comment by fernandocamargoti** [comment]
Done, @menshikh-iv.

**Comment by menshikh-iv** [comment]
@fernandocamargoti nice work, congratz with first contribution :+1:
